### DISCUSSION TITLE
Loader improvements

### DISCUSSION THREAD

**Discussion by wpoortman** [magewirephp/magewire#99] [Q&A] [answered]
I've been working on a overhaul of the loader for ages. Slowly progressing and really liking the current result giving developers much more freedom to finetune specific elements to the shops needs.

Finally I've arrived at the finetuning stage where I would like to throw some last minute decisions into the mix and see what would work best asking you.

So first thing is that you can now split you titles into multiple lines:

![image](https://github.com/magewirephp/magewire/assets/5383956/c0634984-83e5-4497-83b8-f596e7a89435)

Failure style:
![image](https://github.com/magewirephp/magewire/assets/5383956/e39d6ec4-c615-4f26-8726-fdddf0d0923c)

This is done via:
```php
protected $loader = [
    'create' => 'Shipping::New Address::Loading form'
];
```

![image](https://github.com/magewirephp/magewire/assets/5383956/f05f5041-4fd6-41e2-b27e-a39acbfe1476)

This is done via:
```php
protected $loader = [
    'create' => 'Preparing form'
];
```
As you can see, the styling will adapt based on the amount of lines.

So with this syntax, you can split lines using double colons. I'm not fully sure about this, where it could also become a array:
```php
protected $loader = [
    'create' => [
        'Shipping',
        'New Address',
        'Loading form',
    ]
];
```

Why did I use the syntax, because in the future there will be more features relying on a certain syntax. Those are still in the pipeline and won't make it into the next release. And maybe I should skip them, simply not sure yet.

Anyway, let me know what you think and like always all suggestions are welcome. 😉 

WIP branch: https://github.com/magewirephp/magewire/tree/improvements/loader

**Comment by sprankhub** [comment, marked as answer]
Absolutely go for the array syntax, because it is a proper data structure and more flexible :) Plus, it would be impossible to use `::` in the text when you go for the first syntax.

**Reply by wpoortman** [reply]
Thanks Simon! I was struggling with it myself but in the end this is a lot clearer clearer.
